pkg/aws/eni/instances.go

```diff
@@ -105,6 +105,29 @@ func (m *InstancesManager) GetSubnets(ctx context.Context) ipamTypes.SubnetMap {
 	return subnetsCopy
 }
 
+// FindSubnetByIDs returns the subnet with the most addresses matching VPC ID,
+// availability zone within a provided list of subnet ids
+//
+// The returned subnet is immutable so it can be safely accessed
+func (m *InstancesManager) FindSubnetByIDs(vpcID, availabilityZone string, subnetIDs []string) (bestSubnet *ipamTypes.Subnet) {
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+
+	for _, s := range m.subnets {
+		if s.VirtualNetworkID == vpcID && s.AvailabilityZone == availabilityZone {
+			for _, subnetID := range subnetIDs {
+				if s.ID == subnetID {
+					if bestSubnet == nil || bestSubnet.AvailableAddresses < s.AvailableAddresses {
+						bestSubnet = s
+					}
+				}
```

Comment:
Nitpick: We could maybe insert a `continue` here, to avoid looping on the remaining elements in `subnetIDs` after we have found the one with the `subnetID` matching `s.ID`?

```suggestion
					}
					continue
				}
```

Reply:
absolutely :100: